src/machine_learning/perceptron/single_layer_perceptron.rs

```diff
@@ -0,0 +1,103 @@
+// Perceptron model implementation.
+// A perceptron is a basic unit of a neural network, capable of binary classification.
+// It takes inputs, applies weights, and passes the result through an activation function to produce an output.
+
+use crate::machine_learning::perceptron::ActivationFunction;
+use rand::prelude::*;
+
+pub struct Perceptron {
+    weights: Vec<f64>,                 // Weights assigned to each input feature
+    learning_rate: f64,                // Learning rate for weight adjustment during training
+    activation_fn: ActivationFunction, // Activation function used to compute the output
+}
+
+impl Perceptron {
+    // Initialize a new perceptron with random weights, learning rate, and activation function
+    pub fn new(input_size: usize, learning_rate: f64, activation_fn: ActivationFunction) -> Self {
```

Comment:
What do you think about having such signature:
```suggestion
    pub fn new(input_size: usize, learning_rate: f64, activation_fn: F) -> Self 
    where
         F: Fn(f64) -> f64,
    {
```
Instead of the name `F` you can use `ActivationFunction`.
There is not need to have the class `ActivationFunction`.